### PR TITLE
[Merged by Bors] - feat: cleanup file on derivative of Fourier transform

### PR DESCRIPTION
**Pull request by sgouezel**
Rename `mul_L` to `fourierSMulRight` for predictability, expand the file-level docstring, extract a few lemmas from proofs, change `fourier_integral` to `fourierIntegral` in a few lemma names. There is essentially no new mathematical content. This is a preparation for the study of higher order derivatives in #11776.

---

I haven't deprecated the definitions and lemmas in the file `FourierTransformDeriv` when I have changed their names, since as far as I can tell they are not used elsewhere yet. If you think it's important to deprecate them properly, tell me and I'll do it.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 1]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
 /-!
 # Derivative of the Fourier transform
 
-In this file we compute the Fréchet derivative of `𝓕 f`, where `f` is a function such that both
-`f` and `v ↦ ‖v‖ * ‖f v‖` are integrable. Here `𝓕` is understood as an operator `(V → E) → (W → E)`,
-where `V` and `W` are normed `ℝ`-vector spaces and the Fourier transform is taken with respect to a
-continuous `ℝ`-bilinear pairing `L : V × W → ℝ`.
+In this file we compute the Fréchet derivative of the Fourier transform of `f`, where `f` is a
+function such that both `f` and `v ↦ ‖v‖ * ‖f v‖` are integrable. Here the Fourier transform is
+understood as an  operator `(V → E) → (W → E)`, where `V` and `W` are normed `ℝ`-vector spaces
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:15]:
```suggestion
understood as an operator `(V → E) → (W → E)`, where `V` and `W` are normed `ℝ`-vector spaces
```

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 2]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
+* `VectorFourier.fourierSMulRight L f`: given `f : V → E` and `L` a bilinear pairing
+  between `V` and `W`, then this is the function `fun v ↦ -(2 * π * I) (L v ⬝) • f v`,
+  from `V` to `Hom (W, E)`.
+  This is essentially `ContinousLinearMap.smulRight`, up to the factor `- 2πI` designed to make sure
+  that the fourier integral of `fourierSMulRight L f` is the derivative of the Fourier
+  integral of `f`.
 
-We also give a separate lemma for the most common case when `V = W = ℝ` and `L` is the obvious
-multiplication map.
+With this definition, the statements read as follow, first in a general context
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:34]:
```suggestion
With this definition, the statements read as follows, first in a general context
```

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 3]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
+(arbitrary `L` and `μ`):
+
+* `VectorFourier.hasFDerivAt_fourierIntegral`: the Fourier integral of `f` is differentiable, with
+    derivative the Fourier integral of `fourierSMulRight L f`.
+* `VectorFourier.differentiable_fourierIntegral`: the Fourier integral of `f` is differentiable.
+* `VectorFourier.deriv_fourierIntegral`: formula for the derivative of the Fourier integral of `f`.
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:40]: Should this be
```suggestion
* `VectorFourier.fderiv_fourierIntegral`: formula for the derivative of the Fourier integral of `f`.
```

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 4]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
+and the Fourier transform is taken with respect to a continuous `ℝ`-bilinear
+pairing `L : V × W → ℝ` and a given reference measure `μ`.
+
+We give specialized versions of these results on inner product spaces (where `L` is the scalar
+product) and on the real line, where we express the one-dimensional derivative in more concrete
+terms, as the Fourier transform of `x * f x`.
+
+## Main definitions and results
+
+We introduce one handy definition:
```
Comment (ocfnash) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:25]:
```suggestion
We introduce one convenience definition:
```

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 5]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
+
+* `VectorFourier.fourierSMulRight L f`: given `f : V → E` and `L` a bilinear pairing
+  between `V` and `W`, then this is the function `fun v ↦ -(2 * π * I) (L v ⬝) • f v`,
+  from `V` to `Hom (W, E)`.
+  This is essentially `ContinousLinearMap.smulRight`, up to the factor `- 2πI` designed to make sure
+  that the fourier integral of `fourierSMulRight L f` is the derivative of the Fourier
```
Comment (ocfnash) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:31]: I don't have strong feelings on capitalisation but I guess we should be consistent:
```suggestion
  that the Fourier integral of `fourierSMulRight L f` is the derivative of the Fourier
```

File: Mathlib/Analysis/Fourier/FourierTransformDeriv.lean [hunk 6]
```diff
@@ -6,23 +6,55 @@ Authors: Alex Kontorovich, David Loeffler, Heather Macbeth
...
 -/
 
 noncomputable section
 
 open Real Complex MeasureTheory Filter TopologicalSpace
 
-open scoped FourierTransform Topology
+open scoped FourierTransform Topology BigOperators
+
+attribute [local instance 2000] secondCountableTopologyEither_of_left
```
Comment (ocfnash) [inline Mathlib/Analysis/Fourier/FourierTransformDeriv.lean:57]: This priority is fairly opaque to me: is there an explanation that could fit in a comment?

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**